app/src/marketplace/modules/myPurchaseList/actions.js

```diff
@@ -59,3 +76,58 @@ export const getMyPurchases = () => (dispatch: Function) => {
             dispatch(getMyPurchasesFailure(error))
         })
 }
+
+const isSubscriptionActive = (subscription?: ProductSubscription): boolean => isActive((subscription && subscription.endsAt) || '')
+
+const filterPurchases = (data: Array<ProductSubscription>, filter: ?Filter) => {
+    const filtered = data.filter((sub) => {
```

Comment:
We could check if `filter` is present and skip the entire filtering if it \*isn't\*. Example:

```js
if (!filter) {
  return data
}
```

Reply:
Yeah, I totally agree. I'll add this so it's clearer although we have to keep the `filter &&` checks in the code since Flow doesn't get that we already checked if filter is null.